### PR TITLE
Sophos UTM - add module to handle utm ca host key certificate entries

### PR DESCRIPTION
**Pull request by MatrixCrawler**
##### SUMMARY
This PR will add a module to the sophos utm namespace to handle certificate entries.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
- utm_ca_host_key_cert

### CODE REVIEW

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+description:
+    - Create, update or destroy a ca host_key_cert entry in SOPHOS UTM.
+    - This module needs to have the REST Ability of the UTM to be activated.
+
+version_added: "2.8"
+
+options:
+    name:
+        description:
+          - The name of the object. Will be used to identify the entry
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py]:
```suggestion
          - The name of the object. Will be used to identify the entry.
```

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        description:
+          - A reference to an existing utm_ca_signing_ca or utm_ca_verification_ca object.
+        required: true
+    meta:
+        description:
+          - A reference to an existing utm_ca_meta_x509 object.
+        required: true
+    certificate:
+        description:
+          - The certificate in PEM format
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py]:
```suggestion
          - The certificate in PEM format.
```

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        required: true
+    comment:
+        description:
+          - Optional comment string
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py]:
```suggestion
          - Optional comment string.
```

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    encrypted:
+        description:
+          - Optionally enable encryption
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py]:
```suggestion
          - Optionally enable encryption.
```

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        default: False
+        type: bool
+    key:
+        description:
+          - Optional private key in PEM format
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py]:
```suggestion
          - Optional private key in PEM format.
```

File: lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    key_to_check_for_changes = ["ca", "certificate", "comment", "encrypted", "key", "meta"]
+    module = UTMModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            ca=dict(type='str', required=True),
+            meta=dict(type='str', required=True),
+            certificate=dict(type='str', required=True),
+            comment=dict(type='str', required=False),
+            encrypted=dict(type='bool', required=False, default=False),
+            key=dict(type='str', required=False),
```
Comment (Akasurde) [inline lib/ansible/modules/web_infrastructure/sophos_utm/utm_ca_host_key_cert.py]:
```suggestion
            key=dict(type='str', required=False, no_log=True),
```

**Review by Akasurde — COMMENTED** [review]
Rest LGTM

**Review by Akasurde — APPROVED** [review]

**Comment by MatrixCrawler** [comment]
@getjack Will you approve this?

**Comment by maunzCache** [comment]
Sorry have been inactive during holidays. I'll take a look.

**Review by maunzCache — APPROVED** [review]
Seems fine to me. +1

**Comment by Akasurde** [comment]
@MatrixCrawler On a site note, can you please add unit test for UTM modules ? Thanks.

**Pull request merged**